### PR TITLE
Feature/admin list UI

### PR DESCRIPTION
**Pull request by oyakovytskyi**
## Describe your changes

- Created Admin List Page: I developed a new page dedicated to displaying the Admin List.

- Extracted the Admin List Table into a Component: I refactored the Admin List table into a separate component, improving modularity.

- Created a useAdminList Hook: I implemented a custom useAdminList hook to handle the page logic.

- Added Text to i18n: I incorporated all text into the i18n files to support internationalization and easy future translations.

- Separated Styles into a CSS File: All styles were moved to a standalone CSS file, ensuring cleaner component code.

- Standardized Fonts and Colors with Constants: I used constants for fonts and colors to maintain consistency across the application.

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://codecraftersintership.atlassian.net/jira/software/projects/SCRUM/boards/1?selectedIssue=SCRUM-15)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

photo_2024-11-08_22-41-24

### CODE REVIEW

File: src/pages/AdminList/useAdminList.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  paginatedAdmins: Admin[];
+  pageCount: number;
+  startIndex: number;
+  endIndex: number;
+  getInitials: (name: string) => string;
+}
+
+const useAdminList = (): UseAdminListReturnType => {
+  const [admins] = useState<Admin[]>(
+    Array.from({ length: 50 }, (_, index) => ({
```
Comment (lyudad) [inline src/pages/AdminList/useAdminList.ts:31]: add comment //TODO add real data when BE will be ready

File: src/pages/AdminList/useAdminList.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  const pageCount = Math.ceil(filteredAdmins.length / itemsPerPage);
+
+  const getInitials = (name: string): string => {
+    const nameParts = name.split(' ');
+    const initials =
+      nameParts.length >= 2
+        ? `${nameParts[0][0]}${nameParts[1][0]}`
+        : nameParts[0][0];
+    return initials.toUpperCase();
+  };
```
Comment (lyudad) [inline src/pages/AdminList/useAdminList.ts]: I think this function already exist in Utils. If no move it to utils as this used on multiple pages

File: src/pages/AdminList/useAdminList.ts (hunk 1 above)
Comment (oyakovytskyi) [reply src/pages/AdminList/useAdminList.ts:31]: done

File: src/pages/AdminList/useAdminList.ts (hunk 2 above)
Comment (oyakovytskyi) [reply src/pages/AdminList/useAdminList.ts]: made changes

File: src/pages/AdminList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    startIndex,
+    endIndex,
+  } = useAdminList();
+
+  return (
+    <Stack direction={{ xs: 'column', sm: 'row' }} spacing={2} height="100%">
+      <SideBox>
+        <AdminSidebar
+          companyName={t('adminList.companyName')}
+          clientName="John Doe"
```
Comment (lyudad) [inline src/pages/AdminList/index.tsx]: 1.3.3

File: src/pages/AdminList/index.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+          clientEmail="vaftog@vultulkir.org"
```
Comment (lyudad) [inline src/pages/AdminList/index.tsx]: 1.3.3 
if you do mock one you need put in on constant and left comment for other

Comment (oyakovytskyi) [reply src/pages/AdminList/index.tsx]: ok, understand

File: src/pages/AdminList/index.tsx (hunk 3 above)
Comment (oyakovytskyi) [reply src/pages/AdminList/index.tsx]: done

**Pull request merged**